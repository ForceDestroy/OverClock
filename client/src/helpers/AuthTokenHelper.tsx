/* eslint-disable consistent-return */
export default class AuthTokenHelper {
  static getFromLocalStorage = (): string | null => {
    try {
      if (localStorage.getItem('JWT_OC') === null) {
        throw ReferenceError();
      }
      const token = localStorage.getItem('JWT_OC');

      return token;
    } catch (error: unknown) {
      return null;
    }
  };

  static saveToLocalStorage = (token: string, timestamp?: string) => {
    try {
      localStorage.setItem('JWT_OC', token);
      if (timestamp) {
        localStorage.setItem('JWT_OC_TIMESTAMP', timestamp);
      } else {
        localStorage.setItem(
          'JWT_OC_TIMESTAMP',
          new Date().getTime().toString()
        );
      }
    } catch (error: unknown) {
      return null;
    }
  };

  static updateInLocalStorage = (token: string, timestamp?: string) => {
    try {
      if (localStorage.getItem('JWT_OC') === null) {
        throw ReferenceError();
      }
      localStorage.removeItem('JWT_OC');
      localStorage.setItem('JWT_OC', token);
      localStorage.removeItem('JWT_OC_TIMESTAMP');
      if (timestamp) {
        localStorage.setItem('JWT_OC_TIMESTAMP', timestamp);
      } else {
        localStorage.setItem(
          'JWT_OC_TIMESTAMP',
          new Date().getTime().toString()
        );
      }
    } catch (error: unknown) {
      return null;
    }
  };

  static removeFromLocalStorage = () => {
    try {
      if (localStorage.getItem('JWT_OC') === null) {
        throw ReferenceError();
      }
      localStorage.removeItem('JWT_OC');
      localStorage.removeItem('JWT_OC_TIMESTAMP');
      localStorage.removeItem('JWT_OC_NAME');
      localStorage.removeItem('JWT_OC_ACCESS_LEVEL');
      localStorage.removeItem('JWT_OC_USER_ID');
    } catch (error: unknown) {
      return null;
    }
    return null;
  };

  static validateToken = (): boolean => {
    try {
      if (localStorage.getItem('JWT_OC') === null) {
        throw ReferenceError();
      }
      const tokenTimestamp = localStorage.getItem('JWT_OC_TIMESTAMP');
      const timestamp = Date.parse(tokenTimestamp || '');

      if (timestamp + 30 * 60 * 1000 < new Date().getTime()) {
        this.removeFromLocalStorage();
        return false;
      }

      return true;
    } catch (error: unknown) {
      return false;
    }
  };

  static setName = (name: string) => {
    try {
      if (localStorage.getItem('JWT_OC') === null) {
        throw ReferenceError();
      }
      localStorage.setItem('JWT_OC_NAME', name);
    } catch (error: unknown) {
      return null;
    }
  };

  static getName = (): string | null => {
    try {
      if (localStorage.getItem('JWT_OC') === null) {
        throw ReferenceError();
      }
      const name = localStorage.getItem('JWT_OC_NAME');

      return name;
    } catch (error: unknown) {
      return null;
    }
  };

  static setAccessLevel = (accessLevel: string) => {
    try {
      if (localStorage.getItem('JWT_OC') === null) {
        throw ReferenceError();
      }
      localStorage.setItem('JWT_OC_ACCESS_LEVEL', accessLevel);
    } catch (error: unknown) {
      return null;
    }
  };

  static getAccessLevel = (): string | null => {
    try {
      if (localStorage.getItem('JWT_OC') === null) {
        throw ReferenceError();
      }
      const accessLevel = localStorage.getItem('JWT_OC_ACCESS_LEVEL');

      return accessLevel;
    } catch (error: unknown) {
      return null;
    }
  };

  static setUserId = (userId: string) => {
    try {
      if (localStorage.getItem('JWT_OC') === null) {
        throw ReferenceError();
      }
      localStorage.setItem('JWT_OC_USER_ID', userId);
    } catch (error: unknown) {
      return null;
    }
  };

  static getUserId = (): string | null => {
    try {
      if (localStorage.getItem('JWT_OC') === null) {
        throw ReferenceError();
      }
      const userId = localStorage.getItem('JWT_OC_USER_ID');

      return userId;
    } catch (error: unknown) {
      return null;
    }
  };
}
