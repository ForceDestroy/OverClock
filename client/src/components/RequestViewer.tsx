import { useState, useEffect } from 'react';
import { Box, Text, DataTable } from 'grommet';

import AuthTokenHelper from '../helpers/AuthTokenHelper';
import RestCallHelper from '../helpers/RestCallHelper';
import NetworkResources from '../resources/NetworkResources';

function RequestViewer(props: any) {
  const { type } = props;
  const [employeeRequests, setEmployeeRequests] = useState<any[]>([]);
  const formatDate = (date: any) => {
    const year = date.slice(0, 4);
    const month = date.slice(5, 7);
    const day = date.slice(8, 10);
    const hour = date.slice(11, 13);
    const minute = date.slice(14, 16);
    return `${year}-${month}-${day} ${hour}:${minute}`;
  };
  const requestColumns = [
    {
      property: 'date',
      header: 'Date',
      render: (data: any) => <Text>{data.date.slice(0, 10)}</Text>,
    },
    {
      property: 'reason',
      header: 'Reason',
      render: (data: any) => (
        <Text>{data.body.length === 0 ? 'No reason' : data.body}</Text>
      ),
    },
    {
      property: 'start',
      header: 'Start',
      render: (data: any) => <Text>{formatDate(data.startTime)}</Text>,
    },
    {
      property: 'end',
      header: 'End',
      render: (data: any) => <Text>{formatDate(data.endTime)}</Text>,
    },
    {
      property: 'status',
      header: 'Status',
      render: (data: any) => (
        <Text
          color={
            // eslint-disable-next-line no-nested-ternary
            data.status === 'Pending'
              ? 'orange'
              : // eslint-disable-next-line no-nested-ternary
              data.status === 'Approved'
              ? 'green'
              : data.status === 'Acknowledged'
              ? 'blue'
              : 'red'
          }
        >
          {data.status}
        </Text>
      ),
    },
  ];

  const requestCustomColumns = [
    {
      property: 'date',
      header: 'Date',
      render: (data: any) => <Text>{data.date.slice(0, 10)}</Text>,
    },
    {
      property: 'reason',
      header: 'Reason',
      render: (data: any) => (
        <Text>{data.body.length === 0 ? 'No reason' : data.body}</Text>
      ),
    },
    {
      property: 'status',
      header: 'Status',
      render: (data: any) => (
        <Text
          color={
            // eslint-disable-next-line no-nested-ternary
            data.status === 'Pending'
              ? 'orange'
              : // eslint-disable-next-line no-nested-ternary
              data.status === 'Approved'
              ? 'green'
              : data.status === 'Acknowledged'
              ? 'blue'
              : 'red'
          }
        >
          {data.status}
        </Text>
      ),
    },
  ];

  const getEmployeeRequests = async () => {
    const token = AuthTokenHelper.getFromLocalStorage();
    const getRequestsUrl = `${NetworkResources.API_URL}/Request/GetOwnRequests`;
    const getRequestsParams = JSON.stringify({
      sessionToken: token,
    });
    const { data } = await RestCallHelper.GetRequest(
      getRequestsUrl,
      getRequestsParams
    );
    if (data != null) {
      try {
        const dataObject = JSON.parse(data);
        const request: any[] = [];
        for (let i = 0; i < dataObject.length; i += 1) {
          if (dataObject[i].type === type) {
            request.push(dataObject[i]);
          }
        }
        setEmployeeRequests(request);
      } catch (error) {
        console.log(error);
      }
    }
  };

  useEffect(() => {
    getEmployeeRequests();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return (
    <Box>
      <DataTable
        columns={type !== null ? requestColumns : requestCustomColumns}
        data={employeeRequests}
        step={10}
        size="large"
        sortable
      />
    </Box>
  );
}
export default RequestViewer;
