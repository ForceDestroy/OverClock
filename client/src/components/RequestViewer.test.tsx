import { render, screen, waitFor, cleanup } from '@testing-library/react';
import axios from 'axios';
import { describe, it, vi } from 'vitest';

import AuthTokenHelper from '../helpers/AuthTokenHelper';
import RequestViewer from './RequestViewer';

vi.mock('axios');
vi.mock('../helpers/AuthTokenHelper');
describe('RequestViewer', () => {
  beforeEach(() => {
    vi.clearAllMocks();
    cleanup();
  });
  afterEach(() => {
    vi.clearAllMocks();
    vi.resetAllMocks();
  });
  it('Renders RequestViewer to match snapshot', async () => {
    // Mock
    const requestScheduleMock = {
      data: null,
      status: 404,
      statusText: 'Not OK',
    };
    (axios.request as jest.Mock).mockReturnValue(requestScheduleMock);

    (AuthTokenHelper.validateToken as jest.Mock).mockReturnValue(
      'VerySecretToken'
    );
    // Arrange
    const snap = render(<RequestViewer />);

    // Assert
    expect(snap).toMatchSnapshot();
  });
  it('Renders RequestViewer', async () => {
    // Mock
    const requestScheduleMock = {
      data: null,
      status: 404,
      statusText: 'Not OK',
    };
    (axios.request as jest.Mock).mockReturnValue(requestScheduleMock);

    (AuthTokenHelper.validateToken as jest.Mock).mockReturnValue(
      'VerySecretToken'
    );
    render(<RequestViewer />);
    // Assert
    expect(screen.getByText('Date')).toBeInTheDocument();
    expect(screen.getByText('Reason')).toBeInTheDocument();
    expect(screen.getByText('Start')).toBeInTheDocument();
    expect(screen.getByText('End')).toBeInTheDocument();
    expect(screen.getByText('Status')).toBeInTheDocument();
  });
  it('Renders RequestViewer for Pending status', async () => {
    // Mock
    const requestScheduleMock = {
      data: [
        {
          id: 25,
          fromId: 'WEB_000011',
          toId: 'WEB_000010',
          body: '',
          date: '2023-01-17T00:00:00',
          startTime: '2023-01-03T00:00:00',
          endTime: '2023-01-03T00:00:00',
          type: 'Schedule',
          status: 'Pending',
        },
      ],
      status: 200,
      statusText: 'OK',
    };
    (axios.request as jest.Mock).mockReturnValue(requestScheduleMock);

    (AuthTokenHelper.validateToken as jest.Mock).mockReturnValue(
      'VerySecretToken'
    );

    // Arrange
    render(<RequestViewer type="Schedule" />);
    // Assert
    await waitFor(() => {
      expect(screen.getByText('Pending')).toBeInTheDocument();
    });
  });
  it('Renders RequestViewer for Approved status', async () => {
    // Mock
    const requestScheduleMock = {
      data: [
        {
          id: 25,
          fromId: 'WEB_000011',
          toId: 'WEB_000010',
          body: 'I want to change my schedule',
          date: '2023-01-17T00:00:00',
          startTime: '2023-01-03T00:00:00',
          endTime: '2023-01-03T00:00:00',
          type: 'Schedule',
          status: 'Approved',
        },
      ],
      status: 200,
      statusText: 'OK',
    };
    (axios.request as jest.Mock).mockReturnValue(requestScheduleMock);

    (AuthTokenHelper.validateToken as jest.Mock).mockReturnValue(
      'VerySecretToken'
    );

    // Arrange
    render(<RequestViewer type="Schedule" />);

    // Assert
    await waitFor(() => {
      expect(screen.getByText('Approved')).toBeInTheDocument();
      // expect text to not be in the document
      expect(screen.queryByText('No reason')).not.toBeInTheDocument();
    });
  });
  it('Renders RequestViewer for Denied status', async () => {
    // Mock
    const requestScheduleMock = {
      data: [
        {
          id: 25,
          fromId: 'WEB_000011',
          toId: 'WEB_000010',
          body: '',
          date: '2023-01-17T00:00:00',
          startTime: '2023-01-03T00:00:00',
          endTime: '2023-01-03T00:00:00',
          type: 'Schedule',
          status: 'Denied',
        },
      ],
      status: 200,
      statusText: 'OK',
    };
    (axios.request as jest.Mock).mockReturnValue(requestScheduleMock);

    (AuthTokenHelper.validateToken as jest.Mock).mockReturnValue(
      'VerySecretToken'
    );
    // Arrange
    render(<RequestViewer type="Schedule" />);

    // Assert
    await waitFor(() => {
      expect(screen.getByText('Denied')).toBeInTheDocument();
      expect(screen.getByText('No reason')).toBeInTheDocument();
    });
  });
  it('Renders RequestViewer for Acknowledged status', async () => {
    // Mock
    const requestScheduleMock = {
      data: [
        {
          id: 25,
          fromId: 'WEB_000011',
          toId: 'WEB_000010',
          body: 'I want to change my schedule',
          date: '2023-01-17T00:00:00',
          startTime: '2023-01-03T00:00:00',
          endTime: '2023-01-03T00:00:00',
          type: 'Schedule',
          status: 'Acknowledged',
        },
      ],
      status: 200,
      statusText: 'OK',
    };
    (axios.request as jest.Mock).mockReturnValue(requestScheduleMock);

    (AuthTokenHelper.validateToken as jest.Mock).mockReturnValue(
      'VerySecretToken'
    );
    // Arrange
    render(<RequestViewer type="Schedule" />);
    // Assert
    await waitFor(() => {
      expect(screen.getByText('Acknowledged')).toBeInTheDocument();
    });
  });
});
