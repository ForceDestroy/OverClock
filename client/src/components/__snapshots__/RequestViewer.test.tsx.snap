// Vitest Snapshot v1

exports[`RequestViewer > Renders RequestViewer to match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="StyledBox-sc-13pk1d4-0 iNDmPr"
      >
        <table
          class="StyledTable-sc-1m3u5g-6 YPHTu StyledDataTable-sc-xrlyjm-0 hmFzTO"
        >
          <thead
            class="StyledTable__StyledTableHeader-sc-1m3u5g-4 StyledDataTable__StyledDataTableHeader-sc-xrlyjm-4 khLQeI"
          >
            <tr
              class="StyledTable__StyledTableRow-sc-1m3u5g-2 StyledDataTable__StyledDataTableRow-sc-xrlyjm-2 hSlrxd"
            >
              <th
                class="StyledTable__StyledTableCell-sc-1m3u5g-0 ILkGj StyledDataTable__StyledDataTableCell-sc-xrlyjm-6 dsFDzi"
                scope="col"
              >
                <div
                  class="StyledBox-sc-13pk1d4-0 fqUnkf"
                >
                  <button
                    class="StyledButton-sc-323bzc-0 iIPNKk Header__StyledHeaderCellButton-sc-1baku5q-0 jdyvfX"
                    type="button"
                  >
                    <div
                      class="StyledBox-sc-13pk1d4-0 cwefyf"
                    >
                      <span
                        class="StyledText-sc-1sadyjn-0 hpCRgD"
                      >
                        Date
                      </span>
                    </div>
                  </button>
                </div>
              </th>
              <th
                class="StyledTable__StyledTableCell-sc-1m3u5g-0 ILkGj StyledDataTable__StyledDataTableCell-sc-xrlyjm-6 dsFDzi"
                scope="col"
              >
                <div
                  class="StyledBox-sc-13pk1d4-0 fqUnkf"
                >
                  <button
                    class="StyledButton-sc-323bzc-0 iIPNKk Header__StyledHeaderCellButton-sc-1baku5q-0 jdyvfX"
                    type="button"
                  >
                    <div
                      class="StyledBox-sc-13pk1d4-0 cwefyf"
                    >
                      <span
                        class="StyledText-sc-1sadyjn-0 hpCRgD"
                      >
                        Reason
                      </span>
                    </div>
                  </button>
                </div>
              </th>
              <th
                class="StyledTable__StyledTableCell-sc-1m3u5g-0 ILkGj StyledDataTable__StyledDataTableCell-sc-xrlyjm-6 dsFDzi"
                scope="col"
              >
                <div
                  class="StyledBox-sc-13pk1d4-0 fqUnkf"
                >
                  <button
                    class="StyledButton-sc-323bzc-0 iIPNKk Header__StyledHeaderCellButton-sc-1baku5q-0 jdyvfX"
                    type="button"
                  >
                    <div
                      class="StyledBox-sc-13pk1d4-0 cwefyf"
                    >
                      <span
                        class="StyledText-sc-1sadyjn-0 hpCRgD"
                      >
                        Start
                      </span>
                    </div>
                  </button>
                </div>
              </th>
              <th
                class="StyledTable__StyledTableCell-sc-1m3u5g-0 ILkGj StyledDataTable__StyledDataTableCell-sc-xrlyjm-6 dsFDzi"
                scope="col"
              >
                <div
                  class="StyledBox-sc-13pk1d4-0 fqUnkf"
                >
                  <button
                    class="StyledButton-sc-323bzc-0 iIPNKk Header__StyledHeaderCellButton-sc-1baku5q-0 jdyvfX"
                    type="button"
                  >
                    <div
                      class="StyledBox-sc-13pk1d4-0 cwefyf"
                    >
                      <span
                        class="StyledText-sc-1sadyjn-0 hpCRgD"
                      >
                        End
                      </span>
                    </div>
                  </button>
                </div>
              </th>
              <th
                class="StyledTable__StyledTableCell-sc-1m3u5g-0 ILkGj StyledDataTable__StyledDataTableCell-sc-xrlyjm-6 dsFDzi"
                scope="col"
              >
                <div
                  class="StyledBox-sc-13pk1d4-0 fqUnkf"
                >
                  <button
                    class="StyledButton-sc-323bzc-0 iIPNKk Header__StyledHeaderCellButton-sc-1baku5q-0 jdyvfX"
                    type="button"
                  >
                    <div
                      class="StyledBox-sc-13pk1d4-0 cwefyf"
                    >
                      <span
                        class="StyledText-sc-1sadyjn-0 hpCRgD"
                      >
                        Status
                      </span>
                    </div>
                  </button>
                </div>
              </th>
            </tr>
          </thead>
          <tbody
            class="StyledTable__StyledTableBody-sc-1m3u5g-3 StyledDataTable__StyledDataTableBody-sc-xrlyjm-3 iRIncW"
          />
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="StyledBox-sc-13pk1d4-0 iNDmPr"
    >
      <table
        class="StyledTable-sc-1m3u5g-6 YPHTu StyledDataTable-sc-xrlyjm-0 hmFzTO"
      >
        <thead
          class="StyledTable__StyledTableHeader-sc-1m3u5g-4 StyledDataTable__StyledDataTableHeader-sc-xrlyjm-4 khLQeI"
        >
          <tr
            class="StyledTable__StyledTableRow-sc-1m3u5g-2 StyledDataTable__StyledDataTableRow-sc-xrlyjm-2 hSlrxd"
          >
            <th
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 ILkGj StyledDataTable__StyledDataTableCell-sc-xrlyjm-6 dsFDzi"
              scope="col"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 fqUnkf"
              >
                <button
                  class="StyledButton-sc-323bzc-0 iIPNKk Header__StyledHeaderCellButton-sc-1baku5q-0 jdyvfX"
                  type="button"
                >
                  <div
                    class="StyledBox-sc-13pk1d4-0 cwefyf"
                  >
                    <span
                      class="StyledText-sc-1sadyjn-0 hpCRgD"
                    >
                      Date
                    </span>
                  </div>
                </button>
              </div>
            </th>
            <th
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 ILkGj StyledDataTable__StyledDataTableCell-sc-xrlyjm-6 dsFDzi"
              scope="col"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 fqUnkf"
              >
                <button
                  class="StyledButton-sc-323bzc-0 iIPNKk Header__StyledHeaderCellButton-sc-1baku5q-0 jdyvfX"
                  type="button"
                >
                  <div
                    class="StyledBox-sc-13pk1d4-0 cwefyf"
                  >
                    <span
                      class="StyledText-sc-1sadyjn-0 hpCRgD"
                    >
                      Reason
                    </span>
                  </div>
                </button>
              </div>
            </th>
            <th
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 ILkGj StyledDataTable__StyledDataTableCell-sc-xrlyjm-6 dsFDzi"
              scope="col"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 fqUnkf"
              >
                <button
                  class="StyledButton-sc-323bzc-0 iIPNKk Header__StyledHeaderCellButton-sc-1baku5q-0 jdyvfX"
                  type="button"
                >
                  <div
                    class="StyledBox-sc-13pk1d4-0 cwefyf"
                  >
                    <span
                      class="StyledText-sc-1sadyjn-0 hpCRgD"
                    >
                      Start
                    </span>
                  </div>
                </button>
              </div>
            </th>
            <th
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 ILkGj StyledDataTable__StyledDataTableCell-sc-xrlyjm-6 dsFDzi"
              scope="col"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 fqUnkf"
              >
                <button
                  class="StyledButton-sc-323bzc-0 iIPNKk Header__StyledHeaderCellButton-sc-1baku5q-0 jdyvfX"
                  type="button"
                >
                  <div
                    class="StyledBox-sc-13pk1d4-0 cwefyf"
                  >
                    <span
                      class="StyledText-sc-1sadyjn-0 hpCRgD"
                    >
                      End
                    </span>
                  </div>
                </button>
              </div>
            </th>
            <th
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 ILkGj StyledDataTable__StyledDataTableCell-sc-xrlyjm-6 dsFDzi"
              scope="col"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 fqUnkf"
              >
                <button
                  class="StyledButton-sc-323bzc-0 iIPNKk Header__StyledHeaderCellButton-sc-1baku5q-0 jdyvfX"
                  type="button"
                >
                  <div
                    class="StyledBox-sc-13pk1d4-0 cwefyf"
                  >
                    <span
                      class="StyledText-sc-1sadyjn-0 hpCRgD"
                    >
                      Status
                    </span>
                  </div>
                </button>
              </div>
            </th>
          </tr>
        </thead>
        <tbody
          class="StyledTable__StyledTableBody-sc-1m3u5g-3 StyledDataTable__StyledDataTableBody-sc-xrlyjm-3 iRIncW"
        />
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
